openmc/stats/multivariate.py

```diff
@@ -0,0 +1,543 @@
+from abc import ABCMeta, abstractmethod
+from collections.abc import Iterable
+from math import pi
+from numbers import Real
+import sys
+from xml.etree import ElementTree as ET
+
+import numpy as np
+
+import openmc.checkvalue as cv
+from openmc._xml import get_text
+from openmc.stats.univariate import Univariate, Uniform
+
+
+class UnitSphere(metaclass=ABCMeta):
+    """Distribution of points on the unit sphere.
+
+    This abstract class is used for angular distributions, since a direction is
+    represented as a unit vector (i.e., vector on the unit sphere).
+
+    Parameters
+    ----------
+    reference_uvw : Iterable of float
+        Direction from which polar angle is measured
+
+    Attributes
+    ----------
+    reference_uvw : Iterable of float
+        Direction from which polar angle is measured
+
+    """
+    def __init__(self, reference_uvw=None):
+        self._reference_uvw = None
+        if reference_uvw is not None:
+            self.reference_uvw = reference_uvw
+
+    @property
+    def reference_uvw(self):
+        return self._reference_uvw
+
+    @reference_uvw.setter
+    def reference_uvw(self, uvw):
+        cv.check_type('reference direction', uvw, Iterable, Real)
+        uvw = np.asarray(uvw)
+        self._reference_uvw = uvw/np.linalg.norm(uvw)
+
+    @abstractmethod
+    def to_xml_element(self):
+        return ''
+
+    @classmethod
+    @abstractmethod
+    def from_xml_element(cls, elem):
+        distribution = get_text(elem, 'type')
+        if distribution == 'mu-phi':
+            return PolarAzimuthal.from_xml_element(elem)
+        elif distribution == 'isotropic':
+            return Isotropic.from_xml_element(elem)
+        elif distribution == 'monodirectional':
+            return Monodirectional.from_xml_element(elem)
+
+
+class PolarAzimuthal(UnitSphere):
+    """Angular distribution represented by polar and azimuthal angles
+
+    This distribution allows one to specify the distribution of the cosine of
+    the polar angle and the azimuthal angle independently of one another. The
+    polar angle is measured relative to the reference angle.
+
+    Parameters
+    ----------
+    mu : openmc.stats.Univariate
+        Distribution of the cosine of the polar angle
+    phi : openmc.stats.Univariate
+        Distribution of the azimuthal angle in radians
+    reference_uvw : Iterable of float
+        Direction from which polar angle is measured. Defaults to the positive
+        z-direction.
+
+    Attributes
+    ----------
+    mu : openmc.stats.Univariate
+        Distribution of the cosine of the polar angle
+    phi : openmc.stats.Univariate
+        Distribution of the azimuthal angle in radians
+
+    """
+
+    def __init__(self, mu=None, phi=None, reference_uvw=[0., 0., 1.]):
```

Comment:
Note the danger of using mutable default arguments. It doesn't look like you'll run into trouble as it stands, but check out this [explanation](https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html) and consider using a stand in value for the default value of `reference_uvw` (or use a tuple as you did in Spatial methods).